src/explorer/editors/FileEditor.ts

```diff
@@ -36,13 +34,19 @@ export class FileEditor extends BaseEditor<FileTreeItem> {
         return 0;
     }
 
-    public async updateData(node: FileTreeItem): Promise<string> {
-        if (!vscode.window.activeTextEditor) {
-            throw new Error('Cannot update file after it has been closed.');
-        }
-        const localFile: Readable = fs.createReadStream(vscode.window.activeTextEditor.document.uri.fsPath);
+    public async updateData(node: FileTreeItem, data: string): Promise<string> {
         let etag: string = nonNullValue(this._etags.get(node.fullId), 'etag');
-        etag = await putFile(node.root.client, localFile, node.path, etag);
+        try {
+            await putFile(node.root.client, data, node.path, etag);
+        } catch (error) {
+            const parsedError: IParsedError = parseError(error);
+            if (parsedError.message === 'ETag does not represent the latest state of the resource.') {

```

Comment:
This is too dependent on the exact error message. It could change by one letter or be different for other languages. One or both of these checks would be better IMO:
```
parsedError.errorType === '412' && /etag/i.test(parsedError.message)
```